### PR TITLE
AVL Tree

### PR DESCRIPTION
**Pull request by ChinthanBD**
Took chatGPT's help :(

Output:

```
C:\Users\cdevaraju\AppData\Local\Programs\Python\Python39\python.exe C:/Users/cdevaraju/Desktop/123/code/CustomDS/avl_tree.py
AVL Tree:
        -> 80
    -> 70
        -> 60
-> 50
        -> 40
    -> 30
        -> 20

Process finished with exit code 0
```

### CODE REVIEW

**Comment by tejas-k3** [comment]
Can u explain the code?

File: avl_tree.py
Comment (tejas-k3) [inline avl_tree.py:1]: What is the purpose of starting methods with `_` ??

Comment (ChinthanBD) [reply avl_tree.py:1]: functions starting with _ mainly indicates those are kind of private functions and not to expose them. Those functions are only used internally.

And for first question yes, I made sure I understood the code

Comment (tejas-k3) [reply avl_tree.py:1]: In python, are private methods actually private??
Why private methods are introduced here & not in other data structures ?

Comment (ChinthanBD) [reply avl_tree.py:1]: No, the methods are not actually private. We can still access the functions. It is just an indication.
The reason why it was not used in other data structures is that I got to know about this notation while implementing this

Comment (tejas-k3) [reply avl_tree.py:1]: So new learning!!!
try __ for methods and variables in python 2.7 and python 3.10+ and put result in discussion section

**Pull request merged**